app/preprints/discover/route.ts

```diff
@@ -0,0 +1,11 @@
+import Store from '@ember-data/store';
+import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
+
+export default class PreprintDiscoverRoute extends Route {
+    @service store!: Store;
+
+    async model(args: any) {
+        return await this.store.findRecord('preprint-provider', args.provider_id);
```

Comment:
Would it be worthwhile doing a try-catch here and rerouting them if they enter an invalid preprint provider? Maybe it's a little early to implement that though